unit_tests/recovery/main.cpp

```diff
@@ -233,12 +238,295 @@ TEST_CASE ("log recovery parallel test: idle status", "[ci][dbg]")
+  log_lsa_vec.push_back (values_generator.increment_and_get_lsa_log ());
+  auto log_lsa_vec_it = log_lsa_vec.cbegin ();
+
+  SECTION ("1. idle test; will immediately finish")
+  {
+    std::thread observing_thread ([&] ()
+    {
+      const log_lsa min_lsa = min_log_lsa.wait_past_target_lsa (target_log_lsa);
+      REQUIRE (min_lsa != NULL_LSA);
+      REQUIRE (min_lsa == MAX_LSA);
+    });
+    observing_thread.join ();
+  }
+
+  SECTION ("2. produce & consume lsa's; leave others untouched")
+  {
+    // push one value such that we can launch a waiting thread
+    // that will not return immediately
+    min_log_lsa.set_for_produce (*log_lsa_vec_it);
+    ++log_lsa_vec_it;
+
+    std::thread observing_thread ([&] ()
+    {
+      const log_lsa min_lsa = min_log_lsa.wait_past_target_lsa (target_log_lsa);
+      REQUIRE (min_lsa != NULL_LSA);
+      REQUIRE (min_lsa != MAX_LSA);
+    });
+
+    for ( ; ; )
+      {
+	min_log_lsa.set_for_produce (*log_lsa_vec_it);
+	++log_lsa_vec_it;
+	if (log_lsa_vec_it == log_lsa_vec.cend ())
+	  {
+	    break;
+	  }
+
+	// leave in-progress untouched
+	min_log_lsa.set_for_consume_and_in_progress (*log_lsa_vec_it, MAX_LSA);
+	++log_lsa_vec_it;
+	if (log_lsa_vec_it == log_lsa_vec.cend ())
+	  {
+	    break;
+	  }
+
+	const auto current_min_lsa = min_log_lsa.get ();
+	REQUIRE (!current_min_lsa.is_null ());
+	REQUIRE (current_min_lsa != MAX_LSA);
+      }
+    observing_thread.join ();
+    REQUIRE (min_log_lsa.get () != NULL_LSA);
+    REQUIRE (min_log_lsa.get () > target_log_lsa);
+    REQUIRE (min_log_lsa.get () != MAX_LSA);
+  }
+}
+
+TEST_CASE ("minimum log lsa: complete test", "[ci][dbg]")
+{
+  // - use a ut_database_values_generator to generate ever-increasing lsa values
+  // - some deques that mirror the 3 separate containers used in actual functioning: prod, cons, in-progress
+  // - for the 'outer' don't use a container, just a single value
+  // - all these 4 need to be synched by the same mutex
+  // - some threads that randomly move the data simiar to how the actual tasks work and also
+  //    update the monitor
+  // - one more thread that continuosly checks correctness
+
+  constexpr log_lsa SENTINEL_LSA { MAX_LSA };
+
+  // we only use the lsa generation from here
+  const ut_database_config database_config { 42, 42, 2.};
+  ut_database_values_generator global_values{ database_config };
+
+  using log_lsa_deque = std::deque<log_lsa>;
+
+  cublog::minimum_log_lsa_monitor min_log_lsa_monitor;
+
+  log_lsa_deque produce_lsa_deq;
+  log_lsa_deque consume_lsa_deq;
+  log_lsa_deque in_progress_lsa_deq;
+  // seed with a first value
+  log_lsa outer_lsa = global_values.increment_and_get_lsa_log ();
+  min_log_lsa_monitor.set_for_outer (outer_lsa);
+
+  std::mutex deque_mutex;
+
+  std::atomic_bool do_execute_test { true };
+  std::atomic_bool do_execute_test_check { true };
+
+  std::thread checking_thread ([&] ()
+  {
+    while (true)
```

Comment:
Why not include do_execute_test_check directly in the loop condition?
```suggestion
    while (do_execute_test_check)
```